docs/app-spec.md

```diff
@@ -69,6 +69,17 @@ spec:
         # grab only portion of image (optional)
         subPath: inside-dir/dir2
 
+    # pulls content from Docker/OCI registry
```

Comment:
we should drop this into carvel.dev repo.

Reply:
Will add once pr is merged